src/components/GoodLists/GoodList.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import '../GoodLists/GoodList.css';
+import { Buttons } from '../Buttons/Buttons';
+
+const buttonNames = [
+  'Reverse',
+  'Sort alphabetically',
+  'Sort by length',
+  'Reset',
+];
+
+export class GoodList extends React.Component {
+
+  state = {
+    goods: [...this.props.goods],
+    start: false,
+  };
+
+  selectButton = () => {
+    this.setState({ start: true });
+  }
+
+  handleButtonClick = (buttonName) => {
+    if (buttonName === buttonNames[0]) {
+      this.setState((state) => ({
+        goods: [...state.goods.reverse()],
```

Comment:
```suggestion
        goods: [...state.goods].reverse(),
```
you didn't fix the issue from my comment